### ISSUE TITLE
[FEATURE] The Terraform provider for Harvester should provide a bootstrap feature (similar to rancher2) to setup a password and automatically download the configuration

### ISSUE THREAD

**Issue by belgaied2** [harvester/harvester#1797]
**Is your feature request related to a problem? Please describe.**
The Terraform provider for Harvester does not implement a bootstrapping method that can be used after an automatic installation of Harvester. This means, a manual process should happen where a user connects to the UI or a node and gets the KUBECONFIG. This is unfortunate, as the whole process should be more automated.
Suppose someone uses Terraform to provision machines on Equinix (Step 1), they probably want to automatically bootstrap Harvester and continue creating resources inside Harvester (Step 2), without needing to manually connect to a console in between two steps.

**Describe the solution you'd like**
Similarly to the Terraform Rancher2 provider, it should be possible to have a harvester_bootstrap resource (or even use the rancher2 provider) to bootstrap Harvester and output a KUBECONFIG file that can be used by the actual Harvester Terraform configuration.

**Describe alternatives you've considered**
I tried the rancher2_bootstrap resource (now, it does fail with 404 Error code).

**Comment by futuretea** [comment]
Will add Harvester support to https://github.com/rancher/terraform-provider-rancher2

**Comment by lanfon72** [comment]
@futuretea can you provide some test plan?
I can't find any **bootstrap resource** from the PR, and I skimmed those documents, it seems we didn't add any resource to bootstrap Harvester?

**Comment by lanfon72** [comment]
@guangbochen should we verify these resource in the issue?
as they didn't fit the origin feature request, I thought it better be tracking in other feature issue

**Comment by futuretea** [comment]
@lanfon72 @guangbochen  my bad, I misunderstood the feature request at first.

**Comment by futuretea** [comment]
Created another issue: https://github.com/harvester/harvester/issues/2140

**Comment by futuretea** [comment]
If users use terraform-provider-rancher2 to bootstrap the Harvester cluster

for example, use the config as below:
```hcl
terraform {
  required_providers {
    rancher2 = {
      source = "rancher/rancher2"
      version = "1.25.0"
    }
  }
}


provider "rancher2" {
  alias = "bootstrap"

  api_url   = "https://192.168.5.55"
  bootstrap = true
  insecure = true
}

# Create a new rancher2_bootstrap using bootstrap provider config
resource "rancher2_bootstrap" "admin" {
  provider = "rancher2.bootstrap"

  initial_password = "admin"
  password = "blahblah"
  telemetry = true
}


# Provider config for admin
provider "rancher2" {
  alias = "admin"

  api_url = rancher2_bootstrap.admin.url
  token_key = rancher2_bootstrap.admin.token
  insecure = true
}
```

users will got a http 404 error form /v3 api:

```bash
╷
│ Error: [ERROR] Updating token: Bad response statusCode [404]. Status [404 Not Found]. Body: [404 page not found
│ ] from [https://192.168.5.55/v3]
│
│   with rancher2_bootstrap.admin,
│   on test.tf line 20, in resource "rancher2_bootstrap" "admin":
│   20: resource "rancher2_bootstrap" "admin" {
│
╵
```

The reason is we close the Harvester embedded Rancher's MCM feature

If we want to support this bootstrap feature in terraform-provider-harvester, we need to  do it  through kubenetes clients.
- change password: https://github.com/rancher/rancher/blob/61dae8d7e60da7541e5d196569598348053c0ce6/pkg/auth/api/user/user_actions.go#L67
- update settings telemetry-opt
- create settings eula-agreed

**Comment by matttrach** [comment]
This appears to be resolved from the Rancher2 provider side with the merge of #861 , please reopen or generate a new issue if more is necessary.

**Comment by Vicente-Cheng** [comment]
Hi @FrankYang0529, 

What is the current status of this issue?
Is it pending the PR review?

Thanks!

**Comment by FrankYang0529** [comment]
@Vicente-Cheng, yes, the PR is ready for review. I just addressed all comments. Thanks.

**Comment by m-ildefons** [comment]
## Setup

<details>
<summary>Install new Harvester cluster</summary>
<img width="3826" height="2054" alt="Image" src="https://github.com/user-attachments/assets/34d366dd-298d-47a1-af19-a921ee7b8f91" />
</details>

<details>
<summary>Build and install Harvester Terraform provider from git revision
a75a6380b22981ec1997859c3337f279f946a55d</summary>
<img width="789" height="345" alt="Image" src="https://github.com/user-attachments/assets/4d06d7ec-d385-4bfd-87b3-083d86802607" />
</details>

## 🟢 Test 1 -- Initialize Harvester Cluster

<details>
<summary> 🟢 Configure Terraform source with bootstrap resource</summary>
<img width="739" height="758" alt="Image" src="https://github.com/user-attachments/assets/0e2b869d-4648-42f7-aa1d-fa2dec0333d1" />
</details>

<details>
<summary> 🟢 Apply Terraform resources</summary>
<img width="1019" height="849" alt="Image" src="https://github.com/user-attachments/assets/f41c0134-02cf-4be9-a8f5-8de9ccd5a013" />
</details>

## 🟢 Test 2 -- Confirm bootstrap is completed and resource(s) are created as specified

<details>
<summary> 🟢 Bootstrap is completed, admin password is set</summary>
<img width="3829" height="2047" alt="Image" src="https://github.com/user-attachments/assets/0ac52253-6de1-4823-9a22-a1bc72c2074b" />
</details>


<details>
<summary> 🟢 Resources are created as per Terraform source</summary>
<img width="3447" height="547" alt="Image" src="https://github.com/user-attachments/assets/ae1ff8d6-a2ef-4922-89c0-16e610d83a3e" />
</details>

## 🟢 Test 3 -- Confirm re-applying Terraform resources works

<details>
<summary> 🟢 Re-applying Terraform resources without change is successful</summary>
<img width="799" height="90" alt="Image" src="https://github.com/user-attachments/assets/cfd19fe2-57f2-4277-a26f-a1aae70bc5b4" />
</details>

<details>
<summary> 🟢 Re-applying Terraform resources with new password is successful and results in the new password being applied</summary>
<img width="692" height="613" alt="Image" src="https://github.com/user-attachments/assets/c447d561-e88d-4c05-b00a-3c148dcd906b" />
</details>

## Conclusion

 🟢 Validation Passed
